### PR TITLE
Add load_file function

### PR DESCRIPTION
**Pull request by yjw1268**
## What type of PR is this?

- [x] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3418

## What this PR does / why we need it:
Add load_file function

### CODE REVIEW

**Review by sukki37 — CHANGES_REQUESTED** [review]
please add unit test and some bvt cases

**Review by fengttt — CHANGES_REQUESTED** [review]
Nobody other than YueSheng can open a file now.  

Talk to YueSheng, or you the csv loading guy ...

File: pkg/sql/plan/function/builtin/unary/load_file.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	if len(ctx) > blobsize {
+		return nil, errors.New("Data too long for blob")
+	}
+	if err := vec.Append(ctx, proc.GetMheap()); err != nil {
+		return nil, err
+	}
+	return vec, nil
+}
+
+func ReadFromLocalFile(Filepath string) (io.ReadCloser, error) {
```
Comment (fengttt) [inline pkg/sql/plan/function/builtin/unary/load_file.go:50]: Why do you have to read from local?   Can we make Filepath an URL and let fileservice read from local or s3?

**Comment by guguducken** [comment]
@matrix-meow rerun all yes

**Review by fengttt — COMMENTED** [review]
You need a bvt test.

**Review by sukki37 — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

File: pkg/sql/plan/function/builtin/unary/load_file.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/fileservice"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+func LoadFile(vectors []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	inputVector := vectors[0]
+	resultType := types.New(types.T_varchar, 0, 0, 0)
```
Comment (ouyuanning) [inline pkg/sql/plan/function/builtin/unary/load_file.go:30]: why you register return type in builtins.go for T_blob.    set T_varchar here?

File: pkg/sql/plan/function/builtin/unary/load_file.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	resultVector := vector.New(resultType)
+	if inputVector.ConstVectorIsNull() {
+		return vector.NewConstNull(resultType, 1), nil
+	}
+	const blobsize = 65536 // 2^16-1
```
Comment (ouyuanning) [inline pkg/sql/plan/function/builtin/unary/load_file.go:35]: define const variable out of function.

Comment (yjw1268) [reply pkg/sql/plan/function/builtin/unary/load_file.go]: Done.

File: pkg/sql/plan/function/builtin/unary/load_file.go (hunk 2 above)
Comment (yjw1268) [reply pkg/sql/plan/function/builtin/unary/load_file.go]: Fixed that.

**Review by ouyuanning — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Pull request merged**